lib/Dialect/AIEVec/IR/AIEVecOps.cpp

```diff
@@ -1050,6 +1047,102 @@ ParseResult SubOp::parse(OpAsmParser &parser, OperationState &result) {
   return parseAddSubOp(parser, result);
 }
 
+//===----------------------------------------------------------------------===//
+// AddElemOp and SubElemOp
+//===----------------------------------------------------------------------===//
+
+// Print out AddElem and SubElem op.
+template <typename T> void printAddElemSubElemOp(OpAsmPrinter &p, T op) {
+  // Print the lhs operand
+  p << " " << op.getLhs();
+  // Print the rhs operand
+  p << ", " << op.getRhs();
+
+  // Print the types
+  p << " : " << op.getLhs().getType() << ", " << op.getRhs().getType();
+  p << ", " << op.getResult().getType();
+}
+
+void aievec::AddElemOp::print(OpAsmPrinter &p) {
+  printAddElemSubElemOp<aievec::AddElemOp>(p, *this);
+}
+
+void aievec::SubElemOp::print(OpAsmPrinter &p) {
+  printAddElemSubElemOp<aievec::SubElemOp>(p, *this);
+}
+
+// Verify AddElem and SubElem op.
+template <typename T> LogicalResult verifyAddElemSubElemOp(T op) {
+  // Verify the types
+  VectorType resultType =
+      op.getResult().getType().template dyn_cast<VectorType>();
+  VectorType lhsType = op.getLhs().getType().template dyn_cast<VectorType>();
+  VectorType rhsType = op.getRhs().getType().template dyn_cast<VectorType>();
+
+  if (!lhsType || !rhsType || !resultType)
+    return op.emitError("requires vector type");
+
+  // All the vector types must match
+  if (lhsType != rhsType || rhsType != resultType)
+    return op.emitError("all vectors must be of same type");
+
+  return success();
+}
+
+LogicalResult aievec::AddElemOp::verify() {
+  return verifyAddElemSubElemOp<aievec::AddElemOp>(*this);
+}
+
+LogicalResult aievec::SubElemOp::verify() {
+  return verifyAddElemSubElemOp<aievec::SubElemOp>(*this);
+}
+
+// Parse AddElem and SubElem op.
+ParseResult parseAddElemSubElemOp(OpAsmParser &parser, OperationState &result) {
```

Comment:
Is there a reason to be using a custom parser? Can't we use an auto-generated one?